packages/workspace-tools-paths/src/index.ts

```diff
@@ -0,0 +1,48 @@
+import path from "path";
+import fs from "fs";
+import { spawnSync } from "child_process";
+
+/**
+ * Starting from `cwd`, searches up the directory hierarchy for `pathName`.
+ */
+export function searchUp(pathName: string, cwd: string) {
+  const root = path.parse(cwd).root;
+
+  let found = false;
+
+  while (!found && cwd !== root) {
+    if (fs.existsSync(path.join(cwd, pathName))) {
+      found = true;
+      break;
+    }
+
+    cwd = path.dirname(cwd);
+  }
+
+  if (found) {
+    return cwd;
+  }
+
+  return null;
+}
+
+/**
+ * Starting from `cwd`, uses `git rev-parse --show-toplevel` to find the root of the git repo.
+ * Throws if `cwd` is not in a Git repository.
+ */
+export function findGitRoot(cwd: string) {
+  const result = spawnSync("git", ["rev-parse", "--show-toplevel"], { cwd });
```

Comment:
This now uses `spawnSync` directly since the `git()` utility is defined in another package (which consumes this one). I didn't bother with changing the maxBuffer here since the output should be very short.